### PR TITLE
Optionals - Load dynamically based on presence of mod (`__has_include`)

### PR DESCRIPTION
**Pull request by PabstMirror**
ref discussion on https://github.com/acemod/ACE3/pull/8187

pboProject requires a cfgPatch so this adds a dummy one 
(does nothing and is named differently so shouldn't cause problems)

https://feedback.bistudio.com/T161113 is still a problem when using file patching but unlikely to cause issues

### CODE REVIEW

File: .github/workflows/pboproject.yml [hunk 1]
```diff
@@ -35,18 +35,18 @@ jobs:
...
         repository: CBATeam/CBA_A3.git
     - name: Checkout ACE3
-      uses: actions/checkout@v2
+      uses: actions/checkout@master
       with:
         path: z\ace
         persist-credentials: false
     - name: Checkout pull request
-      uses: actions/checkout@v2
+      uses: actions/checkout@master
```
Comment (Dahlgren) [inline .github/workflows/pboproject.yml:49]:
```suggestion
      uses: actions/checkout@v3
```

File: .github/workflows/pboproject.yml [hunk 2]
```diff
@@ -72,13 +72,13 @@ jobs:
       env:
         PYTHONUNBUFFERED: 1
     - name: Archive logs
-      uses: actions/upload-artifact@v2
+      uses: actions/upload-artifact@master
```
Comment (Dahlgren) [inline .github/workflows/pboproject.yml:75]:
```suggestion
      uses: actions/upload-artifact@v3
```

File: tools/.vscode/tasks.json [hunk 3]
```diff
@@ -72,21 +72,23 @@
...
             "group": {
                 "kind": "build",
                 "isDefault": true
             }
         },
         {
-            "label": "Build: Hemtt Release",
-            "command": "hemtt.exe",
+            "label": "Build: Hemtt New",
+            "command": "hemttNew.exe",
```
Comment (Dahlgren) [inline tools/.vscode/tasks.json:85]: This change in binary is not reflected in any docs, revert?

File: .github/workflows/arma.yml [hunk 4]
```diff
@@ -35,14 +35,13 @@ jobs:
   build:
     runs-on: ubuntu-latest
     steps:
-    - name: Checkout the source code
-      uses: actions/checkout@master
-    - name: Build using HEMTT
-      uses: arma-actions/hemtt@master
-      with:
-        command: build --release --ci
+    - uses: actions/checkout@master
```
Comment (Dahlgren) [inline .github/workflows/arma.yml:38]:
```suggestion
    - uses: actions/checkout@v3
```

File: .github/workflows/arma.yml [hunk 5]
```diff
@@ -35,14 +35,13 @@ jobs:
...
+      name: Setup HEMTT
+    - name: Build
+      run: hemtt build
     - name: Upload Artifact
-      uses: actions/upload-artifact@v2-preview
+      uses: actions/upload-artifact@v3
       with:
-        name: ace3-${{ github.sha }}-nobin
-        path: releases/ace3_*.zip
+        name: ace3-${{ github.sha }}
```
Comment (BrettMayson) [inline .github/workflows/arma.yml:46]: This will still be no bin, as binarize isn't available on Linux

Comment (PabstMirror) [reply .github/workflows/arma.yml:46]:
```suggestion
        name: ace3-${{ github.sha }}-nobin
```

Comment (Dahlgren) [reply .github/workflows/arma.yml]: Should we run it on Windows instead then? Can use same prepare step as the pboproject job to have binarize etc setup

Comment (PabstMirror) [reply .github/workflows/arma.yml]: right now we really just want it for the linting

but we could also use it to have a dev/master full build
we'd also need to enable script compiling

File: .github/workflows/arma.yml [hunk 6]
```diff
@@ -35,14 +35,13 @@ jobs:
...
-      uses: actions/checkout@master
-    - name: Build using HEMTT
-      uses: arma-actions/hemtt@master
-      with:
-        command: build --release --ci
+    - uses: actions/checkout@v3
+    - uses: arma-actions/hemtt@v1
+      name: Setup HEMTT
+    - name: Build
+      run: hemtt build
```
Comment (jonpas) [inline .github/workflows/arma.yml:42]:
```suggestion
      run: hemtt build
    - name: Rename build folder
      run: mv .hemttout/build .hemttout/@ace
```

File: .github/workflows/arma.yml [hunk 7]
```diff
@@ -35,14 +35,13 @@ jobs:
...
     - name: Upload Artifact
-      uses: actions/upload-artifact@v2-preview
+      uses: actions/upload-artifact@v3
       with:
         name: ace3-${{ github.sha }}-nobin
-        path: releases/ace3_*.zip
+        path: .hemttout/build
```
Comment (jonpas) [inline .github/workflows/arma.yml:47]:
```suggestion
        path: .hemttout/@*
```

**Review by jonpas — CHANGES_REQUESTED** [review]
Made it directly-loadable by renameing `.hemttout/build` to `.hemttout/@ace` before uploading artifact.

**Comment by PabstMirror** [comment]
artifact doesn't have any bisigns but seems fine otherwise
(compat_ws is not binned but everything else is)

**Review by veteran29 — APPROVED** [review]

**Review by jonpas — APPROVED** [review]

**Pull request merged**